xknx/devices/climate.py

```diff
@@ -165,63 +210,81 @@ def is_on(self):
-        if self.setpoint_shift.value is None:
+        if self._setpoint_shift.value is None:
             return False
         if not self.target_temperature.initialized:
             return False
         if self.target_temperature.value is None:
             return False
         return True
 
+    @property
+    def temperature_step(self):
+        """Return smallest possible temperature step."""
+        if self._setpoint_shift.initialized:
+            return self._setpoint_shift.setpoint_shift_step
+        return DEFAULT_TEMPERATUE_STEP
+
     async def set_target_temperature(self, target_temperature):
-        """Calculate setpoint shift shift and send it to  KNX bus."""
+        """Send new target temperature or setpoint_shift to KNX bus."""
         if self.initialized_for_setpoint_shift_calculations:
-            await self.set_target_temperature_setpoint_shift(target_temperature)
-        # broadcast new target temperature and set internally
-        await self.target_temperature.set(target_temperature)
+            temperature_delta = target_temperature-self.base_temperature
+            await self.set_setpoint_shift(temperature_delta)
+        else:
+            await self.target_temperature.set(target_temperature)
 
-    async def set_target_temperature_setpoint_shift(self, target_temperature):
-        """Set target temperature via setpoint_shift group address."""
-        temperature_delta = target_temperature-self.target_temperature.value
-        setpoint_shift_delta = int(temperature_delta/self.setpoint_shift_step)
-        setpoint_shift = self.setpoint_shift.value + setpoint_shift_delta
-
-        if setpoint_shift > self.setpoint_shift_max:
-            raise DeviceIllegalValue("setpoint_shift_max exceeded", setpoint_shift)
-        if setpoint_shift < self.setpoint_shift_min:
-            raise DeviceIllegalValue("setpoint_shift_min exceeded", setpoint_shift)
+    @property
+    def base_temperature(self):
+        """
+        Return the base temperature.
+
+        Base temperature is the default temperature (setpoint-shift=0) for the active climate mode.
+        As this value is usually not available via KNX, we have to derive this from the current
+        target temperature and the current set point shift.
+        """
+        if self.initialized_for_setpoint_shift_calculations:
+            return self.target_temperature.value - self.setpoint_shift
+        return None
 
-        await self.setpoint_shift.set(setpoint_shift)
+    @property
+    def setpoint_shift(self):
+        """Return current offset from base temperature in Kelvin."""
+        return self._setpoint_shift.value
+
+    async def set_setpoint_shift(self, offset):
+        """Send new temperature offset to KNX bus."""
+        base_temperature = self.base_temperature
+        await self._setpoint_shift.set(offset)
+        # broadcast new target temperature and set internally
+        if self.target_temperature.writable and \
```

Comment:
s/writable/writeable/g

Reply:
Thats what I had initially. Then I had a long google for this. Now I'm unsure again.
Maybe its like tunnelling / tunneling

Reply:
Leave it - both is correct :)